dev/Pager/PrototypePager/PrototypePager.Events.cs

```diff
@@ -192,5 +235,82 @@ private void UpdateHiddenOnEdgeButtons()
             }
         }
 
+        /* Events regarding the number panel display mode */
+
+        private void OnElementPrepared(ItemsRepeater sender, ItemsRepeaterElementPreparedEventArgs args)
+        {
+            if (args.Element == null || args.Element.GetType() != typeof(Button))
+            {
+                return;
+            }
+
+            (args.Element as Button).Click += OnNumberPanelButtonClicked;
+            (args.Element as FrameworkElement).Loaded += MoveIdentifierToCurrentPage;
+        }
+
+        private void OnElementClearing(ItemsRepeater sender, ItemsRepeaterElementClearingEventArgs args)
+        {
+            if (args.Element.GetType() == typeof(Button))
+            {
+                (args.Element as Button).Click -= OnNumberPanelButtonClicked;
+                (args.Element as Button).Loaded -= MoveIdentifierToCurrentPage;
+            }
+        }
+
+        private void OnNumberPanelButtonClicked(object sender, RoutedEventArgs args)
+        {
+            SelectedIndex = (int)(sender as Button).Content;
+        }
+
+        private void MoveIdentifierToCurrentPage(object sender, RoutedEventArgs args)
+        {
+            var element = sender as FrameworkElement;
+
+            if ((int)element.Tag == SelectedIndex)
+            {
+                var boundingRect = LayoutInformation.GetLayoutSlot(element);
+                var numberPanelRectMargins = NumberPanelCurrentPageIdentifier.Margin;
+                NumberPanelCurrentPageIdentifier.Margin = new Thickness(boundingRect.Left, numberPanelRectMargins.Top, numberPanelRectMargins.Right, numberPanelRectMargins.Bottom);  
+            }
+        }
+
+        private void MoveIdentifierToCurrentPage()
+        {
+            for (int i = 0; i < NumberOfPages; i++)
+            {
+                var element = PagerNumberPanel.TryGetElement(i);
+                if (element != null)
+                {
+                    MoveIdentifierToCurrentPage(element, null);
+                }
+            }
+        }
+
+        private void UpdateNumberPanel()
+        {
+            if (NumberOfPages != PagerNumberPanel.ItemsSourceView.Count)
+            {
+                InitializeNumberPanel();
+            }
+
+            if (NumberOfPages <= 7) // Show all pages
```

Comment:
>7 [](start = 33, length = 1)

Avoid having magic number in the code. Consider creating a private static variable to store these values.

Reply:
Done.